dbms/src/Storages/KVStore/MultiRaft/Disagg/CheckpointIngestInfo.cpp

```diff
@@ -193,7 +209,32 @@ bool CheckpointIngestInfo::forciblyClean(TMTContext & tmt, UInt64 region_id, boo
     Page page = uni_ps->read(page_id, nullptr, snapshot, /*throw_on_not_exist*/ false);
     if (unlikely(page.isValid()))
     {
-        CheckpointIngestInfo::removeFromLocal(tmt, region_id);
+        removeFromLocal(tmt, region_id);
+        return true;
+    }
+    return false;
+}
+
+bool CheckpointIngestInfo::forciblyClean(
+    TMTContext & tmt,
+    const TiFlashRaftProxyHelper * proxy_helper,
+    UInt64 region_id)
+{
+    auto uni_ps = tmt.getContext().getWriteNodePageStorage();
+    auto snapshot = uni_ps->getSnapshot(fmt::format("read_fap_i_{}", region_id));
+    auto page_id
+        = UniversalPageIdFormat::toLocalKVPrefix(UniversalPageIdFormat::LocalKVKeyType::FAPIngestInfo, region_id);
+    // For most cases, ingest infos are deleted in `removeFromLocal`.
+    Page page = uni_ps->read(page_id, nullptr, snapshot, /*throw_on_not_exist*/ false);
+    auto maybe_checkpoint_info = CheckpointIngestInfo::restore(tmt, proxy_helper, region_id, 0);
+    if (maybe_checkpoint_info.has_value())
```

Comment:
```suggestion
    // For most cases, ingest infos are deleted in `removeFromLocal`.
    auto maybe_checkpoint_info = CheckpointIngestInfo::restore(tmt, proxy_helper, region_id, 0);
    if (maybe_checkpoint_info.has_value())
```

Are these lines useless?

Reply:
fixed